temporal-sdk/src/test/java/io/temporal/client/functional/WorkflowIdConflictPolicyTest.java

```diff
@@ -0,0 +1,196 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.client.functional;
+
+import static org.junit.Assert.*;
+
+import io.temporal.api.common.v1.WorkflowExecution;
+import io.temporal.api.enums.v1.WorkflowIdConflictPolicy;
+import io.temporal.client.*;
+import io.temporal.testing.internal.SDKTestOptions;
+import io.temporal.testing.internal.SDKTestWorkflowRule;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.shared.TestWorkflows;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class WorkflowIdConflictPolicyTest {
+  @Rule
+  public SDKTestWorkflowRule testWorkflowRule =
+      SDKTestWorkflowRule.newBuilder().setWorkflowTypes(TestWorkflowImpl2.class).build();
+
+  @Test
+  public void policyTerminateExisting() {
+    String workflowId = UUID.randomUUID().toString();
+
+    WorkflowOptions.Builder workflowOptionsBuilder =
+        SDKTestOptions.newWorkflowOptionsWithTimeouts(testWorkflowRule.getTaskQueue()).toBuilder()
+            .setWorkflowIdConflictPolicy(
+                WorkflowIdConflictPolicy.WORKFLOW_ID_CONFLICT_POLICY_TERMINATE_EXISTING)
+            .setWorkflowId(workflowId);
+
+    TestWorkflows.TestSignaledWorkflow workflow1 =
+        testWorkflowRule
+            .getWorkflowClient()
+            .newWorkflowStub(
+                TestWorkflows.TestSignaledWorkflow.class, workflowOptionsBuilder.build());
+    WorkflowStub workflowStub1 = WorkflowStub.fromTyped(workflow1);
+    WorkflowExecution workflowExecution1 = workflowStub1.start();
+    assertNotNull(workflowStub1.getExecution());
+
+    TestWorkflows.TestSignaledWorkflow workflow2 =
+        testWorkflowRule
+            .getWorkflowClient()
+            .newWorkflowStub(
+                TestWorkflows.TestSignaledWorkflow.class, workflowOptionsBuilder.build());
+    WorkflowStub workflowStub2 = WorkflowStub.fromTyped(workflow2);
+    workflowStub2.start();
+    assertNotNull(workflowStub2.getExecution());
+
+    // WORKFLOW_ID_CONFLICT_POLICY_TERMINATE_EXISTING means that calling start with a workflow ID
+    // that already has a running workflow will terminate the existing execution.
+    assertNotEquals(workflowStub1.getExecution(), workflowStub2.getExecution());
+    workflow2.signal("test");
+    assertEquals("done", workflowStub2.getResult(String.class));
+    assertThrows(
+        WorkflowFailedException.class,
+        () ->
+            testWorkflowRule
+                .getWorkflowClient()
+                .newUntypedWorkflowStub(
+                    workflowExecution1,
+                    Optional.of(TestWorkflows.TestSignaledWorkflow.class.toString()))
+                .getResult(String.class));
+  }
+
+  @Test
+  public void policyUseExisting() {
```

Comment:
Looks like this test fails against server (as it should). We should probably just hold up this PR until the code is in server.